src/widget/form/addfriendform.h

```diff
@@ -55,9 +55,9 @@ class AddFriendForm : public QObject
     bool addFriendRequest(const QString& friendAddress, const QString& message);
 
 signals:
-    void friendRequested(const QString& friendAddress, const QString& message);
-    void friendRequestAccepted(const QString& friendAddress);
-    void friendRequestsSeen();
+    void friendRequested(const QString& friendAddress, const QString& message) const;
+    void friendRequestAccepted(const QString& friendAddress) const;
+    void friendRequestsSeen() const;
```

Comment:
No, it doesn't! A const method is meant to not alter anything within it's context, right? So any method or signal called inside a const method is enforced to be const as well. If a const method emits a signal, something's wrong! Why, because nothing changed actually! Breaking this so-called const-correctness by declaring non-const methods const will result in a lot of `const_cast<type>()` casts as a consequence!